aten/src/ATen/native/quantized/cpu/qlinear_dynamic.cpp

```diff
@@ -232,11 +233,10 @@ at::Tensor PackedLinearWeightsQnnp::apply_dynamic_impl(at::Tensor input) {
   size_t rows_w = bias_.size(0);
   size_t cols_w = input_contig.size(input_contig.dim() - 1);
 
-  at::Tensor bias_vec = bias_;
+  TORCH_CHECK(bias_.dim() == 1, "bias should be a vector (1D Tensor)");
 
-  TORCH_CHECK(bias_vec.dim() == 1, "bias should be a vector (1D Tensor)");
-
-  auto bias_contig = bias_vec.contiguous();
+  auto bias_contig = at::native::xnnpack::internal::allocate_padded_contiguous_if_needed(
+      bias_, bias_.suggest_memory_format());
```

Comment:
This is the least intrusive fix I could think of that solves issue and checks the performance boxes (no reallocation on mobile), but is ugly in that it's adding a header dependency on XNNPACK integration.  Ideally we would want that file to be in a common location.  Do you want me to address that here or in a separate commit (or hopefully someone else can refactor this since I need to spend some time on GPU.)

Reply:
Also technically this causes an extra allocation on non-mobile if the number of output channels is already a multiple of 8 in which case a reallocation is not needed.